src/emulator/util/CMakeLists.txt

```diff
@@ -1,11 +1,17 @@
 add_library(
 	util
 	STATIC
+	include/util/BitField.h
+	include/util/types.h
```

Comment:
I'm really against adding a `types.h` header as it's way too nebulous and guaranteed to grow into a junkyard.

Reply:
I'm also inclined to move source code appropriated from RPCS3 to src/external, as then we don't have to try and make it match in style or formatting, etc.

Reply:
All other utility files use it though... that's the main reason I added it.
I can try stripping it off to absolutely necessary stuff, or idk, splitting it into new files with appropriate names?